### PR TITLE
PHP 8 Compatiblity trial - adding PHP 8.0

### PR DESCRIPTION
**Pull request by bobstrecansky**
Making the changes necessary to be PHP 8.0 compatible.  Comments in line.

### CODE REVIEW

File: .github/workflows/php.yml [hunk 1]
```diff
@@ -44,6 +44,8 @@ jobs:
       run: composer install --prefer-dist --no-progress --no-suggest
 
     - name: Check Style
+      env:
+        PHP_CS_FIXER_IGNORE_ENV: 1
```
Comment (bobstrecansky) [inline .github/workflows/php.yml:48]: I had to add this to bypass the environment check for PHP8 based on [this comment](https://github.com/FriendsOfPHP/PHP-CS-Fixer/issues/4702#issuecomment-738086179)

I'm planning on removing this enviornment variable once php-cs-fixer is more generally available for PHP8.

File: .phan/config.php [hunk 2]
```diff
@@ -366,7 +366,7 @@
         'vendor/php-http',
         'vendor/phan/phan/src/Phan',
         'vendor/phpunit/phpunit/src',
-        'vendor/endclothing/prometheus_client_php/src',
+        'vendor/promphp/prometheus_client_php/src',
```
Comment (bobstrecansky) [inline .phan/config.php:369]: The endclothing version of this library stopped receiving support.  The promphp version is the continued support fork of this library:

https://github.com/endclothing/prometheus_client_php/issues/56

File: composer.json [hunk 3]
```diff
@@ -4,12 +4,12 @@
     "type": "library",
     "license": "Apache-2.0",
     "require": {
-        "php": "^7.3",
+        "php": "^7.3 || ^8.0",
         "ext-json": "*",
-        "guzzlehttp/guzzle": "^6.5",
+        "guzzlehttp/guzzle": "^7.1.0",
```
Comment (bobstrecansky) [inline composer.json:9]: We had to upgrade to guzzle 7.x.x for PHP compatibility.

File: composer.json [hunk 4]
```diff
@@ -4,12 +4,12 @@
...
         "psr/http-client": "^1.0",
-        "php-http/guzzle6-adapter": "^2.0",
-        "endclothing/prometheus_client_php": "^1.0"
+        "php-http/guzzle7-adapter": "^0.1.1",
```
Comment (bobstrecansky) [inline composer.json:11]: Same with the guzzle7-adapter in re: PHP8 compatibility.

File: docker/Dockerfile [hunk 5]
```diff
@@ -1,10 +1,10 @@
-FROM php:7.3-buster
+FROM php:8-buster
 
 RUN apt-get -y update && apt-get -y install git zip && \
 curl -sS https://getcomposer.org/installer | php && \
 mv composer.phar /usr/local/bin/composer && \
 chmod +x /usr/local/bin/composer && \
-pecl install ast-1.0.4 xdebug && \
+pecl install ast-1.0.10 xdebug && \
```
Comment (bobstrecansky) [inline docker/Dockerfile:7]: `ast-1.0.4` wouldn't compile for PHP8, `ast-1.0.10` works for all three versions of php (`7.3`, `7.4`, and `8.0`)

**Review by beniamin — APPROVED** [review]

**Pull request merged**